Content.Server/Atmos/Piping/Unary/EntitySystems/GasCanisterSystem.cs

```diff
@@ -44,6 +45,24 @@ public override void Initialize()
             SubscribeLocalEvent<GasCanisterComponent, GasCanisterHoldingTankEjectMessage>(OnHoldingTankEjectMessage);
             SubscribeLocalEvent<GasCanisterComponent, GasCanisterChangeReleasePressureMessage>(OnCanisterChangeReleasePressure);
             SubscribeLocalEvent<GasCanisterComponent, GasCanisterChangeReleaseValveMessage>(OnCanisterChangeReleaseValve);
+
+        }
+
+        /// <summary>
+        /// Completely dumps the content of the canister into the world.
+        /// </summary>
+        /// <param name="uid"></param>
+        /// <param name="canister"></param>
+        public void PurgeContents(EntityUid uid, GasCanisterComponent canister)
+        {
+            if (EntityManager.TryGetComponent<ITransformComponent>(uid, out var transform))
+            {
+                var environment = _atmosphereSystem.GetTileMixture(transform.Coordinates, true);
+                if (environment is not null)
+                {
+                    _atmosphereSystem.Merge(environment, canister.Air);
```

Comment:
Doing this doesn't mutate the canister's gas mixture, so maybe clear it after this?